controlpanel/frontend/static/javascripts/modules/role-names.js

```diff
@@ -0,0 +1,61 @@
+moj.Modules.roleNames = {
+  id: "role_name",
+  selectId: "role_name-select",
+  selectSelector: "#role_name-select",
+  roleEndpointAttr: "data-role-endpoint",
+  formClass: ".appRoles",
+  autocompleteWrapperClass: ".autocomplete__wrapper",
+  roleFilters: {
+    airflow: new RegExp('^airflow_'),
+    webapp: new RegExp('^(alpha|dev)_app_'),
+  },
+  roles: null,
+
+  init() {
+    this.$selectField = $(this.selectSelector);
+    this.roleListEndpoint = this.$selectField.attr(this.roleEndpointAttr);
+    if (document.querySelectorAll(this.formClass).length) {
+      this.getRoles().done(() => {
+        this.loadRolesToSelect();
+        this.bindEvents();
+      });
+    }
+  },
+
+  bindEvents() {
+    document.querySelectorAll('input[name=app_type]').forEach(input => {
+      input.addEventListener('change', (event) => {
+        this.loadRolesToSelect();
+      });
+    });
+  },
+
+  getRoles() {
+    return $.get(
```

Comment:
You could get rid of JQuery altogether by replacing this with [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Examples), but it's not urgent.